File: sorting/bozosort.py
# encoding: utf-8

"""
Bozosort
Autor: Bozo
Tipo: Ordenação de vetores
Descrição: Embaralha um vetor indefinidamente, até que os números estejam em ordem.
Complexidade:  
    Pior caso: O(infinito)
    Melhor caso: O(n)
Referências:
    http://nerdson.com/blog/libbozo-01/
    http://pt.wikipedia.org/wiki/Bogosort
"""

from random import shuffle

def bozosort(seq):
    como_deve_ficar = sorted(seq)
    while seq != como_deve_ficar:
        shuffle(seq)
    return seq

print bozosort([2,4,9,1,0,-4,17,8,0,23,67,-1])

File: sorting/bozosort/bozosort.php
<?php
	
/*
Bozosort (a.k.a Bogosort)
Autor: 
	Bozo
Colaborador:
    Maurício Sipmann (sipmann@gmail.com)
Tipo: 
	sorting
Descrição: 
	Ordena um array a fim de ficar em ordem crescente.
Complexidade:
	O(infinito)
Dificuldade:
	facil
Referências:
	http://pt.wikipedia.org/wiki/Bogosort
*/

	function bogo($valores) {
		$ordenado = false;
		$certo = $valores;
		sort($certo);
		
		$i = 0;
		
		while(!$ordenado) {
			shuffle($valores);
			if( $valores == $certo) {
				$ordenado = true;
			}
			$i++;
			echo "Tentativa ".$i." <br>";
			print_r(($ordenado==true)? $valores: "");
		}
		
	}
	
	// Passando um array com valores a serem ordenados
	bogo(array(3,5,2,1));
	
?>
